### PR TITLE
FFmpeg 6 support

### PR DESCRIPTION
**Pull request by crsib**
Resolves: #4381 

*(short description of the changes and the motivation to make the changes)*

- [x] I signed [CLA](https://www.audacityteam.org/cla/)
- [x] The title of the pull request describes an issue it addresses
- [x] If changes are extensive, then there is a sequence of easily reviewable commits
- [x] Each commit's message describes its purpose and effects
- [x] There are no behavior changes unnecessary for the stated purpose of the PR

Recommended:
- [x] Each commit compiles and runs on my machine without known undesirable changes of behavior

### CODE REVIEW

File: libraries/lib-ffmpeg-support/CMakeLists.txt [hunk 1]
```diff
@@ -70,6 +70,13 @@ if (${_OPT}use_ffmpeg)
       impl/avformat/59/AVFormatImpl.cpp
       impl/avutil/57/AVUtilImpl.cpp
 
+      impl/ffmpeg-6.0.0-single-header.h
+      impl/avutil/58/avconfig.h
+      impl/avcodec/60/AVCodecIDLookup.cpp
+      impl/avcodec/60/AVCodecImpl.cpp
+      impl/avformat/60/AVFormatImpl.cpp
+      impl/avutil/58/AVUtilImpl.cpp
```
Comment (Paul-Licameli) [inline libraries/lib-ffmpeg-support/CMakeLists.txt:78]: I see a mix of "58" and "60" here which I assume is intended

File: libraries/lib-ffmpeg-support/impl/avcodec/60/AVCodecIDLookup.cpp [hunk 2]
```diff
@@ -0,0 +1,37 @@
+extern "C"
+{
+#include "../../avutil/58/avconfig.h"
```
Comment (Paul-Licameli) [inline libraries/lib-ffmpeg-support/impl/avcodec/60/AVCodecIDLookup.cpp:3]: 60 using 58 again, comment ?

File: libraries/lib-ffmpeg-support/impl/avformat/60/AVFormatImpl.cpp [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+  AVFormatImpl.cpp
+
+  Dmitry Vedenko
+
+**********************************************************************/
+
+extern "C"
+{
+#include "../../avutil/58/avconfig.h"
```
[the remark this answers is not in the document]
Comment (Paul-Licameli) [reply libraries/lib-ffmpeg-support/impl/avformat/60/AVFormatImpl.cpp:13]: Or maybe not? ... maybe this difference of number is just frequent in all the other versions too.  I'm comparing with 55 for instance.

File: libraries/lib-ffmpeg-support/impl/avcodec/60/AVCodecIDLookup.cpp (hunk 2 above)
Comment (crsib) [reply libraries/lib-ffmpeg-support/impl/avcodec/60/AVCodecIDLookup.cpp:3]: This files are literally copied, none of the original files have a comment about different versions between avutil and the rest of the libraries. 

The difference between versions is due to the difference inside the FFmpeg itself.

**Review by Paul-Licameli — APPROVED** [review]

**Pull request merged**